### PR TITLE
omer.faruk/EI-966-Add_VNET_Integration_FA_Module

### PR DESCRIPTION
**Pull request by omerf-hmcts**
omer.faruk/EI-966-Add_VNET_Integration_FA_Module

### CODE REVIEW

**Review by akozichev-hmcts — COMMENTED** [review]
look ok

File: main.tf [hunk 1]
```diff
@@ -24,7 +23,7 @@ data "azurerm_service_plan" "sp" {
 # Function App
 resource "azurerm_linux_function_app" "linux_function" {
   count                       = var.asp_os_type == "Linux" ? 1 : 0
-  name                        = var.function_app_name
+  name                        = "${var.function_app_name}${var.revision}"
```
Comment (varunvijayagopal) [inline main.tf:26]: why are we changing name ?

File: variables.tf [hunk 2]
```diff
@@ -131,3 +131,48 @@ variable "tags" {
...
+}
+
+
+variable "revision" {
+   description = "Revision of Functional App"
+   type        = string
+   default     = ""
+}
+
+variable "namespace" {
```
Comment (varunvijayagopal) [inline variables.tf:167]: don't think this variable is needed for function app

**Review by varunvijayagopal — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (omerf-hmcts) [reply variables.tf:167]: Ok removing namespace..

File: main.tf [hunk 3]
```diff
@@ -81,6 +80,42 @@ resource "azurerm_linux_function_app" "linux_function" {
...
+  resource_group_name  = var.vnet_rg_name
+  depends_on = [
+    azurerm_subnet.main
+  ]
+}
+
+
+resource "azurerm_app_service_virtual_network_swift_connection" "linux" {
+  # count          = var.create_subnet && length(var.subnet_cidr) != 0 && length(var.subnet_name) == 0 && var.asp_os_type == "Linux" ? 1 : 0
+  count          = var.asp_os_type == "Linux" ? 1 : 0
```
Comment (akozichev-hmcts) [inline main.tf:114]: Looks like the connection created unconditionally, but not all FA will need to be network bound. Do you need to use the line above instead?

File: main.tf [hunk 4]
```diff
@@ -139,6 +174,14 @@ resource "azurerm_windows_function_app" "windows_function" {
   }
 }
 
+resource "azurerm_app_service_virtual_network_swift_connection" "windows" {
+  # count          = var.create_subnet && length(var.subnet_cidr) != 0  && length(var.subnet_name) == 0 && var.asp_os_type == "Windows" ? 1 : 0
+  count          = var.asp_os_type == "Windows" ? 1 : 0
```
Comment (akozichev-hmcts) [inline main.tf:179]: same here - probably need to swap comments

**Review by akozichev-hmcts — DISMISSED** [review]

**Review by gunnertwin — APPROVED** [review]

**Review by varunvijayagopal — DISMISSED** [review]
fix terratest

**Review by akozichev-hmcts — APPROVED** [review]
looks ok

**Comment by omerf-hmcts** [comment]
> fix terratest

Done!

**Pull request merged**